service/tablet_allocator.cc

```diff
@@ -1149,6 +1150,36 @@ class load_balancer {
     }
 };
 
+future<tablet_reallocation_result>
+reallocate_tablets_for_new_rf(const tablet_aware_replication_strategy* rep, schema_ptr s, token_metadata_ptr tm,
```

Comment:
A couple of notes regarding the function's signature:
1. I don't really like the name, as it implies we're reallocating the tablets, whereas this function doesn't have any effect on the system, it just calculates new tablets replicas map - I believe I raised this point a couple of times already
2. also, tablets don't get reallocated, some are added, some are removed, but none gets reallocated anywhere
3. I don't know the code that much, but is `s` a common name for the _table_ schema, and not table or keyspace schema? If it's the latter, then it'd be better to name this var more specifically, i.e. whether it's a ks or cf schema.
4. Just by looking at the parameters passed to this function I have hard time figuring out why would they be needed, but I'll just comment on them below. 
5. `ks_options` probably doesn't contain ks options, but only rf per dc mapping, correct? If that's so, it has to be renamed. 
6. Why would only 2 out of 4 params be `const`?

Reply:
1. Yes.
3. OK, thanks for clarifying that.
5. Still, since it's only rf per dc map, I'd name it like that, i.e. `rf_per_dc`.
6. OK, it didn't occur to me these are just pointers.